### PR TITLE
Fix for issue #1538

### PR DESCRIPTION
**Pull request by mbudiu-vmw**
I apologize for the large set of changes to the outputs. I don't know how to avoid those with our current testing framework.

### CODE REVIEW

**Comment by jafingerhut** [comment]
No addition of the test program that exhibits the bug?  Maybe it was in there but has neither the original name nor the name issue1538 so I missed it.

**Comment by mbudiu-vmw** [comment]
I had called it issue1358 instead of 1538.
I have committed a renamed version.

**Comment by mbudiu-vmw** [comment]
Can we get a review for this patch?

File: frontends/p4/uniqueNames.cpp [hunk 1]
```diff
@@ -80,8 +80,8 @@ addNameAnnotation(cstring name, const IR::Annotations* annos) {
 }
 
 UniqueNames::UniqueNames(ReferenceMap* refMap) : renameMap(new RenameMap) {
-    setStopOnError(true);
     setName("UniqueNames");
+    visitDagOnce = false;
```
Comment (ChrisDodd) [inline frontends/p4/uniqueNames.cpp]: Won't this create multiple unique names for a single shared object?  Thus turning it into multiple objects?

**Pull request merged**